test/LinearCurve.test.js

```diff
@@ -0,0 +1,38 @@
+/* global require beforeEach describe it ethers */
+
+const { assert } = require('chai')
+
+describe('LinearCurve', () => {
+  let linearCurve
+
+  beforeEach(async () => {
+    const LinearCurve = await ethers.getContractFactory('LinearCurve')
+    linearCurve = await LinearCurve.deploy()
+  })
+
+  it('returns the correct price', async () => {
+    async function price (supply, amount) {
+      const bigSupply = ethers.constants.WeiPerEther.mul(supply)
```

Comment:
Something I discovered late in my test writing last time is that you can use `ethers.utils.parseEther(supply)` here and that's a little nicer I think.